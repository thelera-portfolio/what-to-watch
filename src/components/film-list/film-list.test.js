import FilmList from "./film-list.jsx";
import React from "react";
import renderer from "react-test-renderer";

const films = [
  {
    image: `img/snatch.jpg`,
    preview: `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`,
    title: `Snatch`,
  },
  {
    image: `img/aviator.jpg`,
    poster: `ddd`,
    preview: `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`,
    title: `Aviator`,
  },
  {
    image: `img/avatar.jpg`,
    preview: `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`,
    title: `Avatar`,
  }
];

it(`FilmList is rendered correctly`, () => {
  const tree = renderer.create(
      <FilmList
        films={films}
        onClick={() => {}}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
