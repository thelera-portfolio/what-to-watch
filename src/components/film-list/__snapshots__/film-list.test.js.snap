// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList is rendered correctly 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="Snatch"
        height="175"
        src="img/snatch.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
      >
        Snatch
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="Aviator"
        height="175"
        src="img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
      >
        Aviator
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="Avatar"
        height="175"
        src="img/avatar.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
      >
        Avatar
      </a>
    </h3>
  </article>,
]
`;
